src/Contracts/OpdexVault.cs

```diff
@@ -44,35 +41,93 @@ public Address Token
     }
 
     /// <inheritdoc />
-    public Address Owner
+    public UInt256 TotalSupply
     {
-        get => State.GetAddress(VaultStateKeys.Owner);
-        private set => State.SetAddress(VaultStateKeys.Owner, value);
+        get => State.GetUInt256(VaultStateKeys.TotalSupply);
+        private set => State.SetUInt256(VaultStateKeys.TotalSupply, value);
     }
 
     /// <inheritdoc />
-    public Address PendingOwner
+    public ulong NextProposalId
     {
-        get => State.GetAddress(VaultStateKeys.PendingOwner);
-        private set => State.SetAddress(VaultStateKeys.PendingOwner, value);
+        get => State.GetUInt64(VaultStateKeys.NextProposalId);
+        private set => State.SetUInt64(VaultStateKeys.NextProposalId, value);
     }
 
     /// <inheritdoc />
-    public UInt256 TotalSupply
+    public UInt256 TotalProposedAmount
     {
-        get => State.GetUInt256(VaultStateKeys.TotalSupply);
-        private set => State.SetUInt256(VaultStateKeys.TotalSupply, value);
+        get => State.GetUInt256(VaultStateKeys.TotalProposedAmount);
+        private set => State.SetUInt256(VaultStateKeys.TotalProposedAmount, value);
+    }
+
+    /// <inheritdoc />
+    public ulong PledgeMinimum
+    {
+        get => State.GetUInt64(VaultStateKeys.PledgeMinimum);
+        private set => State.SetUInt64(VaultStateKeys.PledgeMinimum, value);
+    }
+
+    /// <inheritdoc />
+    public ulong ProposalMinimum
```

Comment:
Wondering if this could have a negative effect and incentivise not voting. For example, a proposal is passing which someone doesn't want to pass. The proposal doesn't have much more weight than the minimum, so rather than voting no they pull their vote causing it to go below the threshold.

Reply:
Thats entirely up to the user to do so if they are not that committed to making sure it passes. I think nothing to do here.